apps/chat/src/components/Common/ApplicationWizard/ApplicationWizardFooter.tsx

```diff
@@ -0,0 +1,159 @@
+import { IconTrashX, IconWorldShare } from '@tabler/icons-react';
+import { FC, useCallback, useState } from 'react';
+
+import { useTranslation } from 'next-i18next';
+
+import classNames from 'classnames';
+
+import { getFolderIdFromEntityId } from '@/src/utils/app/folders';
+import { ApiUtils } from '@/src/utils/server/api';
+
+import { CustomApplicationModel } from '@/src/types/applications';
+import { SharingType } from '@/src/types/share';
+import { Translation } from '@/src/types/translation';
+
+import { ApplicationActions } from '@/src/store/application/application.reducers';
+import { useAppDispatch } from '@/src/store/hooks';
+
+import { PublishModal } from '@/src/components/Chat/Publish/PublishWizard';
+import { ConfirmDialog } from '@/src/components/Common/ConfirmDialog';
+import Tooltip from '@/src/components/Common/Tooltip';
+
+import { PublishActions } from '@epam/ai-dial-shared';
+
+interface ApplicationWizardFooterProps {
+  onClose: (v: boolean) => void;
+  isEdit?: boolean;
+  isValid?: boolean;
+  selectedApplication?: CustomApplicationModel;
+}
+
+export const ApplicationWizardFooter: FC<ApplicationWizardFooterProps> = ({
+  isEdit,
+  isValid,
+  selectedApplication,
+  onClose,
+}) => {
+  const { t } = useTranslation(Translation.Chat);
+
+  const dispatch = useAppDispatch();
+
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+  const [isPublishing, setIsPublishing] = useState(false);
+
+  const applicationToPublish = selectedApplication
```

Comment:
lets add iconUrl also here, it will fix 1 more #1941 case (to publish icons along with apps)